.github/workflows/ci.yml

```diff
@@ -67,6 +67,9 @@ jobs:
       fail-fast: false
       matrix:
         pkg:
+          - name: Arrow.jl (dev'd ArrowTypes.jl)
+            dir: '.'
+            dev: './src/ArrowTypes'
           - name: Arrow.jl
             dir: '.'
```

Comment:
For this PR the tests will only pass when using the local ArrowTypes.jl. However, we probably don't want to have this always be the way we run tests as otherwise we're not testing that the ArrowTypes compat entry is correct. This is a compromise which requires the PR reviewer to be aware of the scenarios in which it's okay for some of these checks to fail.

I happy to remove these changes from these PR or make them into a separate PR. I included them here to demonstrate that the Arrow.jl tests are functional with the latest ArrowTypes.jl

Reply:
Removed the CI changes